stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/preview/ImagePreviewActivity.kt

```diff
@@ -475,52 +494,136 @@ public class ImagePreviewActivity : AppCompatActivity() {
-                    )
-                    .transformable(state = transformableState)
-                    .pointerInput(Unit) {
-                        detectTapGestures(
-                            onDoubleTap = {
-                                val newScale = when {
-                                    currentScale == 3f -> 1f
-                                    currentScale >= 2f -> 3f
-                                    else -> 2f
+            BoxWithConstraints(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.Center
+            ) {
+                val painter = rememberImagePainter(data = attachments[page].imagePreviewUrl)
+
+                val density = LocalDensity.current
+                val parentSize = Size(density.run { maxWidth.toPx() }, density.run { maxHeight.toPx() })
+                var imageSize by remember { mutableStateOf(Size(0f, 0f)) }
+
+                var currentScale by remember { mutableStateOf(1f) }
+                var translation by remember { mutableStateOf(Offset(0f, 0f)) }
+
+                val scale by animateFloatAsState(targetValue = currentScale)
+
+                Image(
+                    modifier = Modifier
+                        .then(
+                            (painter.state as? ImagePainter.State.Success)
+                                ?.painter
+                                ?.intrinsicSize
+                                ?.let { intrinsicsSize ->
+                                    Modifier.aspectRatio(intrinsicsSize.width / intrinsicsSize.height, true)
+                                } ?: Modifier
+                        )
+                        .graphicsLayer(
+                            scaleY = scale,
+                            scaleX = scale,
+                            translationX = translation.x,
+                            translationY = translation.y
+                        )
+                        .onGloballyPositioned {
+                            imageSize = Size(it.size.width.toFloat(), it.size.height.toFloat())
+                        }
+                        .pointerInput(Unit) {
+                            forEachGesture {
+                                awaitPointerEventScope {
+                                    awaitFirstDown(requireUnconsumed = true)
+                                    do {
+                                        val event = awaitPointerEvent(pass = PointerEventPass.Initial)
+
+                                        val maxTranslation = calculateMaxOffset(
+                                            imageSize = imageSize,
+                                            scale = currentScale,
+                                            parentSize = parentSize
+                                        )
+
+                                        val zoom = event.calculateZoom()
+                                        currentScale = (zoom * currentScale).coerceAtMost(3f)
```

Comment:
should pull out as a constant so it's not a magic number (e.g. `DefaultMaxImageZoomScale`)